src/parser/ir/Ir.g4

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2022 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+grammar Ir;
+
+//-----------------------------------------------------------------------------
+// Parser
+//-----------------------------------------------------------------------------
+
+
+attribute
+    : ID ':' NUMLITERAL
+    | ID ':' '"'ID'"'
+    ;
+attributeList
+    : attribute (',' attribute)*
+    ;
+value
+    : ANY
+    | ID
+    | ID'.'ID
+    ;
+argumentList
+    : value (',' value)*
+    ;
+operation
+    : ID '=' ID '['(attributeList)?']''('(argumentList)?')'
+    ;
+irProgram
+    : operation
+    ;
+
+//-----------------------------------------------------------------------------
+// Lexer
+//-----------------------------------------------------------------------------
+
+// keywords
+ANY: '<<ANY>>';
+
+ID : ('%')? [_a-zA-Z0-9/.]*;
+NUMLITERAL : [0-9]+;
+RESULT : '%'[0-9]+;
```

Comment:
I actually think we might need to keep `RESULT` and change it to be `[ID [, ID]*]`. This will be the correct representation for list. This makes it easy to build dependency graph in the future if we have to ensure the correct order of operations.